lib/parser/parse-tree.cc

```diff
@@ -163,13 +168,19 @@ Statement<ActionStmt> StmtFunctionStmt::ConvertToAssignment() {
   auto &funcName{std::get<Name>(t)};
   auto &funcArgs{std::get<std::list<Name>>(t)};
   auto &funcExpr{std::get<Scalar<Expr>>(t).thing};
+  CharBlock source{funcName.source};
   std::list<Expr> subscripts;
   for (Name &arg : funcArgs) {
-    subscripts.push_back(
-        Expr{common::Indirection{Designator{DataRef{Name{arg}}}}});
+    subscripts.push_back(WithSource(arg.source,
+        Expr{common::Indirection{
+            WithSource(arg.source, Designator{DataRef{Name{arg}}})}}));
+    source.ExtendToCover(arg.source);
   }
-  auto variable{Variable{common::Indirection{
-      MakeArrayElementRef(funcName, funcName.source, std::move(subscripts))}}};
+  // extend source to include closing paren
+  CHECK(*source.end() == ')');
+  source = CharBlock{source.begin(), source.end() + 1};
```

Comment:
This is a kludge but I can't find any other way to get the source location of the new Designator.

We are converting the LHS of a stmt-function-stmt (e.g. `f(i) = 1`) into a Variable. There is no source location for the LHS. We could add one during parsing, but it would not be the location for the entire node. So we would probably want to split the StmtFunctionNode into LHS and RHS so we have a place to put the LHS source position. This seems like a lot of trouble to go to just to avoid this kludge.